Build fails with latest docker image (esphome/esphome:dev)

### The problem

Compiling code with latest docker image fails with following error:

`


Compiling .pioenvs/distancemeter-esp32/src/esphome/components/mqtt/mqtt_light.cpp.o
src/esphome/components/mqtt/mqtt_light.cpp:2:56: fatal error: esphome/components/light/light_json_schema.h: No such file or directory
compilation terminated.
*** [.pioenvs/distancemeter-esp32/src/esphome/components/mqtt/mqtt_light.cpp.o] Error 1

`

### What is version of ESPHome has the issue?

latest dev

### What type of installation are you using?

Docker

### What platform are you using?

ESP32

### Board

D1 Mini

### Component causing the issue

src/esphome/components/mqtt/mqtt_light.cpp

### Example YAML snippet

_No response_

### Anything in the logs that might be useful for us?

_No response_

### Additional information

_No response_

--- a/esphome/components/mqtt/mqtt_light.cpp
+++ b/esphome/components/mqtt/mqtt_light.cpp
@@ -1,9 +1,9 @@
 #include "mqtt_light.h"
-#include "esphome/components/light/light_json_schema.h"
 #include "esphome/core/log.h"
 
 #ifdef USE_LIGHT
 
+#include "esphome/components/light/light_json_schema.h"
 namespace esphome {
 namespace mqtt {
 

